tests/terratest/functions/set/setRKE1.go

```diff
@@ -0,0 +1,163 @@
+package functions
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	format "github.com/rancher/rancher/tests/terratest/functions/format"
+	"github.com/rancher/rancher/tests/v2/validation/terratest"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func SetRKE1(k8sVersion string, nodePools []terratest.Nodepool, file *os.File) (done bool, err error) {
+	rancherConfig := new(rancher.Config)
+	config.LoadConfig("rancher", rancherConfig)
+
+	terraformConfig := new(terratest.TerraformConfig)
+	config.LoadConfig("terraform", terraformConfig)
+
+	// create new empty hcl file object
+	f := hclwrite.NewEmptyFile()
+
+	// initialize the body of the new file object
+	rootBody := f.Body()
+
+	// initialize terraform object and set req provider/version
+	tfBlock := rootBody.AppendNewBlock("terraform", nil)
+	tfBlockBody := tfBlock.Body()
+	reqProvsBlock := tfBlockBody.AppendNewBlock("required_providers", nil)
+	reqProvsBlockBody := reqProvsBlock.Body()
+
+	reqProvsBlockBody.SetAttributeValue("rancher2", cty.ObjectVal(map[string]cty.Value{
+		"source":  cty.StringVal("rancher/rancher2"),
+		"version": cty.StringVal(terraformConfig.ProviderVersion),
+	}))
+	rootBody.AppendNewline()
+
+	// Provider
+	provBlock := rootBody.AppendNewBlock("provider", []string{"rancher2"})
+	provBlockBody := provBlock.Body()
+	provBlockBody.SetAttributeValue("api_url", cty.StringVal(`https://`+rancherConfig.Host))
+	provBlockBody.SetAttributeValue("token_key", cty.StringVal(rancherConfig.AdminToken))
+	provBlockBody.SetAttributeValue("insecure", cty.BoolVal(*rancherConfig.Insecure))
+	rootBody.AppendNewline()
+
+	// Resource node template
+	nodeTemplateBlock := rootBody.AppendNewBlock("resource", []string{"rancher2_node_template", "rancher2_node_template"})
+	nodeTemplateBlockBody := nodeTemplateBlock.Body()
+	nodeTemplateBlockBody.SetAttributeValue("name", cty.StringVal(terraformConfig.NodeTemplateName))
+
+	if terraformConfig.Module == "ec2_rke1" {
+		ec2ConfigBlock := nodeTemplateBlockBody.AppendNewBlock("amazonec2_config", nil)
+		ec2ConfigBlockBody := ec2ConfigBlock.Body()
+		ec2ConfigBlockBody.SetAttributeValue("access_key", cty.StringVal(terraformConfig.AWSAccessKey))
+		ec2ConfigBlockBody.SetAttributeValue("secret_key", cty.StringVal(terraformConfig.AWSSecretKey))
+		ec2ConfigBlockBody.SetAttributeValue("ami", cty.StringVal(terraformConfig.Ami))
+		ec2ConfigBlockBody.SetAttributeValue("region", cty.StringVal(terraformConfig.Region))
+		awsSecGroupNames := format.ListOfStrings(terraformConfig.AWSSecurityGroupNames)
```

Comment:
Same nit here about code readability and separating variables out.

Reply:
Are you talking about formatting and spacing? What do you mean separating variables out?